xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocumentArchive.java

```diff
@@ -263,6 +263,9 @@ public void setArchive(String text) throws XWikiException
             Collection nodes = archive.getNodes(getWikiReference(), getId());
             for (Iterator it = nodes.iterator(); it.hasNext();) {
                 XWikiRCSNodeInfo nodeInfo = (XWikiRCSNodeInfo) it.next();
+                if (!it.hasNext()) {
+                    throw new IllegalStateException("The number of nodes is an odd number.");
+                }
```

Comment:
This change is not correct: as you can see at https://github.com/xwiki/xwiki-platform/blob/5fa50d553ccd8c4f5ac7356fbc249df0add3ae17/xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/rcs/XWikiRCSArchive.java#L284-L285 there are always a pair number of nodes.